plugins/modem-manager/fu-mm-device.c

```diff
@@ -1504,6 +1546,34 @@ fu_mm_device_attach(FuDevice *device, FuProgress *progress, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_mm_device_setup(FuDevice *device, GError **error)
+{
+	FuMmDevice *self = FU_MM_DEVICE(device);
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autoptr(GError) error_branch = NULL;
+
+	/* Create IO channel to send AT commands to the modem */
+	locker = fu_device_locker_new_full(device,
+					   (FuDeviceLockerFunc)fu_mm_device_io_open,
+					   (FuDeviceLockerFunc)fu_mm_device_io_close,
+					   error);
+	/*
+	 * firmware branch AT command may fail if not implemented,
+	 * clear error if not supported
+	 */
+	if (self->branch_at != NULL &&
+	    !fu_mm_device_at_cmd(self, self->branch_at, TRUE, error_branch))
+		g_debug("Unable to get firmware branch: %s", error_branch->message);
+
+	if (fu_device_get_branch(device) != NULL)
+		g_debug("Found firmware branch: %s", fu_device_get_branch(device));
+	else
+		g_debug("Running default firmware branch");
```

Comment:
do we really need this?

Reply:
I added to make it clear from the logs which branch is used.
So whatever modem you have, you can always find the branch in the logs.

Reply:
maybe just `g_debug("using firmware branch: %s", fu_device_get_branch(device));` for both cases?

Reply:
That would be `null` then for the default branch which doesn't read great in the logs.

Reply:
okay, fair enough; just make the first word lower case then LGTM.